lib/step.js

```diff
@@ -38,8 +38,39 @@ class Step {
     this.metaStep = undefined;
     /** @member {string} */
     this.stack = '';
-    /** @member {number} */
-    this.totalTimeout = undefined;
+
+    const timeouts = new Map();
+    /**
+     * @method
+     * @returns {number|undefined}
+     */
+    this.getTimeout = function () {
+      let totalTimeout;
+      // iterate over all timeouts starting from highest values of order
+      new Map([...timeouts.entries()].sort().reverse()).forEach((timeout, order) => {
+        if (timeout !== undefined && (
+          // when orders >= 10 - timeout value overrides those set with higher order elements
+          order >= 10
```

Comment:
this breaks my mind... Magical numbers :mage: 🔮
what the point of `10` number here?

Why not just the lower number is is more priority it gets?